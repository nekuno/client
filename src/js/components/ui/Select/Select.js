import PropTypes from 'prop-types';
import React, { Component } from 'react';
import InputRadio from '../InputRadio/InputRadio.js';
import styles from './Select.scss';

export default class Select extends Component {

    static propTypes = {
        title          : PropTypes.string,
        options        : PropTypes.array.isRequired,
        defaultOption  : PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
        onClickHandler : PropTypes.func.isRequired
    };

    constructor(props) {
        super(props);

        this.handleClick = this.handleClick.bind(this);

        this.state = {
            selected: props.defaultOption || null
        };
    }

    handleClick(value) {
        this.setState({selected: value});

        this.props.onClickHandler(value);
    }


    render() {
        const {title, options} = this.props;
        const {selected} = this.state;

        return (
            <div className={styles.selectWrapper}>
                {title ? <div className={styles.title + ' small'}>{title}</div> : null}
                <div className={styles.select}>
                    {options.map((option, key) =>
                        <InputRadio key={key} value={option.key} text={option.text} checked={selected !== option.key} onClickHandler={this.handleClick}/>
                    )}
                </div>
            </div>
        );
    }
}