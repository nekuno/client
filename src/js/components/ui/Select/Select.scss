@import '../../../../scss/_partials/colors';

.select-wrapper {

  .title {
    font-weight: 600;
  }
  .select {
    .option-wrapper {

      .option {

        .optionText {

        }

      }

      &.selected {

        .option {

          .optionText {

          }
        }
      }
    }
  }
}