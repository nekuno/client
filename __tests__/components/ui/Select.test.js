import React from 'react';
import Select from '../../../src/js/components/ui/Select/Select.js';
import styles from '../../../src/js/components/ui/Select/Select.scss';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const options = [
    {
        key: "lorem",
        text: "Lorem"
    },
    {
        key: "ipsum",
        text: "Ipsum"
    },
    {
        key: "lorem-ipsum",
        text: "Lorem ipsum"
    },
    {
        key: "sed-ut-perspiciatis",
        text: "Sed ut perspiciatis"
    }
];

describe('Test Select component', () => {
    const mockFn = jest.fn();
    const select = shallow(<Select title={'Lorem ipsum'} options={options} defaultOption={"lorem"} onClickHandler={mockFn}/>);

    it('should be defined', () => {
        expect(select).toBeDefined();
    });
    it('should render correctly', () => {
        expect(select).toMatchSnapshot();
    });
    // it('Select click event', () => {
    //     select.find(`.${styles.optionWrapper}`).get(0).simulate('click');
    //     expect(mockFn).toHaveBeenCalledTimes(1);
    // });
});
