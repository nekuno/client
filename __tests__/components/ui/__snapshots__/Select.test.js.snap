// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Select component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Select
    defaultOption="lorem"
    onClickHandler={[MockFunction]}
    options={
      Array [
        Object {
          "key": "lorem",
          "text": "Lorem",
        },
        Object {
          "key": "ipsum",
          "text": "Ipsum",
        },
        Object {
          "key": "lorem-ipsum",
          "text": "Lorem ipsum",
        },
        Object {
          "key": "sed-ut-perspiciatis",
          "text": "Sed ut perspiciatis",
        },
      ]
    }
    title="Lorem ipsum"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="title small"
        >
          Lorem ipsum
        </div>,
        <div
          className="select"
        >
          <InputRadio
            checked={false}
            onClickHandler={[Function]}
            text="Lorem"
            value="lorem"
          />
          <InputRadio
            checked={true}
            onClickHandler={[Function]}
            text="Ipsum"
            value="ipsum"
          />
          <InputRadio
            checked={true}
            onClickHandler={[Function]}
            text="Lorem ipsum"
            value="lorem-ipsum"
          />
          <InputRadio
            checked={true}
            onClickHandler={[Function]}
            text="Sed ut perspiciatis"
            value="sed-ut-perspiciatis"
          />
        </div>,
      ],
      "className": "selectWrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem ipsum",
          "className": "title small",
        },
        "ref": null,
        "rendered": "Lorem ipsum",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <InputRadio
              checked={false}
              onClickHandler={[Function]}
              text="Lorem"
              value="lorem"
            />,
            <InputRadio
              checked={true}
              onClickHandler={[Function]}
              text="Ipsum"
              value="ipsum"
            />,
            <InputRadio
              checked={true}
              onClickHandler={[Function]}
              text="Lorem ipsum"
              value="lorem-ipsum"
            />,
            <InputRadio
              checked={true}
              onClickHandler={[Function]}
              text="Sed ut perspiciatis"
              value="sed-ut-perspiciatis"
            />,
          ],
          "className": "select",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "checked": false,
              "onClickHandler": [Function],
              "text": "Lorem",
              "value": "lorem",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "checked": true,
              "onClickHandler": [Function],
              "text": "Ipsum",
              "value": "ipsum",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "class",
            "props": Object {
              "checked": true,
              "onClickHandler": [Function],
              "text": "Lorem ipsum",
              "value": "lorem-ipsum",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "class",
            "props": Object {
              "checked": true,
              "onClickHandler": [Function],
              "text": "Sed ut perspiciatis",
              "value": "sed-ut-perspiciatis",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title small"
          >
            Lorem ipsum
          </div>,
          <div
            className="select"
          >
            <InputRadio
              checked={false}
              onClickHandler={[Function]}
              text="Lorem"
              value="lorem"
            />
            <InputRadio
              checked={true}
              onClickHandler={[Function]}
              text="Ipsum"
              value="ipsum"
            />
            <InputRadio
              checked={true}
              onClickHandler={[Function]}
              text="Lorem ipsum"
              value="lorem-ipsum"
            />
            <InputRadio
              checked={true}
              onClickHandler={[Function]}
              text="Sed ut perspiciatis"
              value="sed-ut-perspiciatis"
            />
          </div>,
        ],
        "className": "selectWrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lorem ipsum",
            "className": "title small",
          },
          "ref": null,
          "rendered": "Lorem ipsum",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <InputRadio
                checked={false}
                onClickHandler={[Function]}
                text="Lorem"
                value="lorem"
              />,
              <InputRadio
                checked={true}
                onClickHandler={[Function]}
                text="Ipsum"
                value="ipsum"
              />,
              <InputRadio
                checked={true}
                onClickHandler={[Function]}
                text="Lorem ipsum"
                value="lorem-ipsum"
              />,
              <InputRadio
                checked={true}
                onClickHandler={[Function]}
                text="Sed ut perspiciatis"
                value="sed-ut-perspiciatis"
              />,
            ],
            "className": "select",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "checked": false,
                "onClickHandler": [Function],
                "text": "Lorem",
                "value": "lorem",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "checked": true,
                "onClickHandler": [Function],
                "text": "Ipsum",
                "value": "ipsum",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "class",
              "props": Object {
                "checked": true,
                "onClickHandler": [Function],
                "text": "Lorem ipsum",
                "value": "lorem-ipsum",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "class",
              "props": Object {
                "checked": true,
                "onClickHandler": [Function],
                "text": "Sed ut perspiciatis",
                "value": "sed-ut-perspiciatis",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
